### PR TITLE
feat(Schedule): support query for listing schedules

### PR DESCRIPTION
**Pull request by nikhilbhatia08**
## What was changed

 Added the support for query for listing schedules and added test coverage on the changes.

## Checklist

1. Closes  [issue](https://github.com/temporalio/sdk-typescript/issues/1457)

2. How was this tested:

Added a test to the above changes as discussed with @antlai-temporal .
Created four schedules with two schedules having a search attribute key and two of them not having search attribute keys and then listing with the key ensuring I got back two of them.
549 tests passed
2 tests skipped
9 tests todo

3. Any docs updates needed?

### CODE REVIEW

File: packages/test/src/test-schedules.ts [hunk 1]
```diff
@@ -542,6 +542,87 @@ if (RUN_INTEGRATION_TESTS) {
...
 
+  test.serial('Can list Schedules with a query string', async (t) => {
+    const { client } = t.context;
+
+    const groupId = randomUUID();
+    const createdScheduleHandlesPromises = [];
+    const expectedIds: string[] = [];
+    for(let i = 0; i < 4; i++) {
+      const scheduleId = `test-query-${groupId}-${i + 1}`
+      if(i < 2) {
```
Comment (mjameswh) [inline packages/test/src/test-schedules.ts:553]: Duplicating the whole block force one to be more attentive to figure out how the branches differ.

Instead, I'd just initialize the `searchAttributes` variable first (either undefined/empty or containing the custom search attribute) and add the schedule id to the list of expected schedules at the same time.

Something like:

```
const searchAttributes: SearchAttributes = {};
if (i < 2) {
  searchAttributes['CustomKeywordField'] = [ ... ];
  expectedIds.push(scheduleId);
}
```

Comment (nikhilbhatia08) [reply packages/test/src/test-schedules.ts:553]: Yes @mjameswh Correcting it right away. Thank you for correcting.

**Comment by nikhilbhatia08** [comment]
@mjameswh removed the duplicating block. Please review it sir.

File: packages/test/src/test-schedules.ts [hunk 2]
```diff
@@ -542,6 +543,72 @@ if (RUN_INTEGRATION_TESTS) {
...
+          scheduleId,
+          spec: {
+            calendars: [{ hour: { start: 2, end: 7, step: 1 } }],
+          },
+          action: {
+            type: 'startWorkflow',
+            workflowType: dummyWorkflow,
+            taskQueue,
+          },
+          searchAttributes: Object.keys(searchAttributes).length ? searchAttributes : undefined,
```
Comment (mjameswh) [inline packages/test/src/test-schedules.ts:570]: Simply that:

```suggestion
          searchAttributes,
```

Comment (nikhilbhatia08) [reply packages/test/src/test-schedules.ts:570]: @mjameswh changing it right away sir.

**Comment by nikhilbhatia08** [comment]
@mjameswh removed the extra code sir. Please review it sir. My apologies for small mistakes and I truly appreciate your patience and understanding sir.

**Comment by mjameswh** [comment]
> My apologies for small mistakes and I truly appreciate your patience and understanding sir.

No worry for that, a few round of these are totally expected. Thanks for your own patience, really, and thanks for your contribution.

**Comment by mjameswh** [comment]
Another minor thing [here](https://github.com/temporalio/sdk-typescript/actions/runs/11073262800/job/30772179175#step:11:19). I believe that should be the last one.

BTW, you can run the linter yourself on your local machine. It's a good habit to run these two commands just before pushing commits to a public branch:

```
npm run format
npm run lint
```

Different Open Source projects have different tools, but most use some kind of linters.

**Comment by nikhilbhatia08** [comment]
@mjameswh sir ran both the commands npm run format and npm run lint.

**Review by mjameswh — APPROVED** [review]
Great! Thanks a lot!

**Pull request merged**